src/cljc/orcpub/pdf_spec.cljc

```diff
@@ -145,6 +145,33 @@
                                          (actions-string "-----------Other Traits------------" traits)]))
                                traits-str))}))
 
+(defn traits-fields [built-char]
+  (let [traits-by-type (group-by :type (es/entity-val built-char :traits))
+        bonus-actions (sort-by :name (concat (es/entity-val built-char :bonus-actions)
+                                             (traits-by-type :b-action)))
+        actions (sort-by :name (concat (es/entity-val built-char :actions)
+                                       (traits-by-type :action)))
+        reactions (sort-by :name (concat (es/entity-val built-char :reactions)
+                                         (traits-by-type :reaction)))
+        traits (sort-by :name (traits-by-type nil))
+        traits-str (traits-string traits)
+        actions? (or (seq bonus-actions)
+                     (seq actions)
+                     (seq reactions)
+                     (seq traits))
+        header (features-and-traits-header built-char)
+        ]
+    {:features-and-traits-3 (str header "\n\n"
```

Comment:
Can't rename :features-and-traits-2 to :features-and-traits-3.
It is the form field name in the PDF that needs to match.

![image](https://github.com/Orcpub/orcpub/assets/3913859/a9cee5d9-e505-4e2e-b591-1701ca54efda)

Testing the other PDF's they come across blank.

Reply:
Sounds good. I found a fallback that I can make whats called a "PDF Portfolio" that will allow multiple of the same form field name while not linking them so I can set Char limits

Reply:
It is strange that they are coming across blank though

Reply:
@Jessomadic 
This work flow takes the form data from the "download-form" in the browser and loops through all the elements in it.

![image](https://github.com/Orcpub/orcpub/assets/3913859/6fca2d98-e93c-4fd6-b3e3-0e016e48d976)

If you look in the data, there is no :features-and-traits-3. But there is :features-and-traits-2. 

It's looping through all the fields being passed down from the as built character.

Ideally, we need a chunk of code that looks at the size of what features-and-traits are going to be and adds another page if needed.

Example, where 900 is the character length that looks good in the PDF Text field. something like:

do while ( features-and-traits >=900 )
add page
take first 900 characters from features-and-traits and insert it into the page.
remove the first 900 characters from features-and-traits and set features-and-traits to the new value.
)

The routes is responsible for filling the PDF text fields, and it is called here.
https://github.com/Orcpub/orcpub/blob/6b709e7912dc06c11afab7758ef5deaf7b2b66cc/src/clj/orcpub/routes.clj#L478

Implementation of it lives here:
https://github.com/Orcpub/orcpub/blob/6b709e7912dc06c11afab7758ef5deaf7b2b66cc/src/clj/orcpub/pdf.clj#L31-L55
Which is just looping through all the fields/elements that are in the edn.

I don't know. It's a toss up to break up that function or just live with it.

Reply:
Just live with it. I'll make a workaround. It was more if it was an easy 10 min change it would be nice to have but I can get it all done in Adobe. Thanks for breaking it down though that explains alot

Reply:
I'll look into the logic zotz pointed out once I get a free weekend again.